include/solvers/mpm_base.tcc

```diff
@@ -96,10 +96,29 @@ mpm::MPMBase<Tdim>::MPMBase(const std::shared_ptr<IO>& io) : mpm::MPM(io) {
   try {
     if (post_process_.at("vtk_statevars").is_array() &&
         post_process_.at("vtk_statevars").size() > 0) {
-      for (unsigned i = 0; i < post_process_.at("vtk_statevars").size(); ++i) {
-        std::string attribute =
-            post_process_["vtk_statevars"][i].template get<std::string>();
-        vtk_statevars_.emplace_back(attribute);
+      // Iterate over state_vars
+      for (const auto& svars : post_process_["vtk_statevars"]) {
+        // Phase id
+        unsigned phase_id = 0;
+        if (svars.contains("phase_id"))
+          phase_id = svars.at("phase_id").template get<unsigned>();
+
+        // State variables
+        std::vector<std::string> state_var;
+        if (svars.at("statevars").is_array() &&
```

Comment:
`std::vector<std::string> state_var = svars["statevars"];`

Reply:
We don't have to iterate

Reply:
Fixed